scripts/zones/Lower_Jeuno/npcs/Treasure_Coffer.lua

```diff
@@ -478,6 +590,77 @@ local function givePrize(player, ki)
+    param2   = bit.rshift(option, 16) - (param3 * 32)                                      -- Second selected augment option
+    param1   = bit.rshift(option, 11) - (param3 * 1024) - (param2 * 32)                    -- First selected augment option
+    gear     = bit.rshift(option, 6)  - (param3 * 32768)- (param2 * 1024) - (param1 * 32)  -- Gear selection
+    addon    = optionToGear[gear].addon
+
+    if addonReward then
+        option = option  - 8388615 -- extra bit passed when recieving armor
+    end
+
+    if addonReward then
+        for i = 1, #parameter1 do                        -- Adding parameter1 augments table
+            table.insert(addAug, parameter1[i].augment)
+            table.insert(addAug, parameter1[i].power)
+        end
+        for i = 1, #parameter2 do                        -- Adding parameter2 augments table
+            table.insert(addAug, parameter2[i].augment)
+            table.insert(addAug, parameter2[i].power)
+        end
+
+        if player:getFreeSlotsCount() == 0 then
+            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED, optionToGear[gear].itemid)
+        else
+            player:addItem(optionToGear[gear].itemid, 1, unpack(addAug))
+            player:messageSpecial(ID.text.ITEM_OBTAINED, optionToGear[gear].itemid)
+            player:delKeyItem(({xi.ki.PRISMATIC_KEY, xi.ki.OXBLOOD_KEY, xi.ki.BEHEMOTH_KEY})[addon])
+        end
+    else
+        for i = 1, #parameter1 do -- Converting augment selection 1 integer values into binary for updateEvent (5 bits for power followed by 11 bits for ID)
+            table.insert(addAug, string.format("%05i%011i", intToBinary(parameter1[i].power), intToBinary(parameter1[i].augment)))
+        end
+        for i = 1, #parameter2 do -- Converting augment selection 2 integer values into binary for updateEvent (5 bits for power followed by 11 bits for ID)
+            table.insert(addAug, string.format("%05i%011i", intToBinary(parameter2[i].power), intToBinary(parameter2[i].augment)))
+        end
+        for i = #addAug, 5 do
+            table.insert(addAug, "0000000000000000")
+        end
+
+        if addon == 1 then -- Printing augment 1 and 2 selections for visuals in binary. The first argument must contact the string below.
+            player:updateEvent(tonumber(addAug[1] .. "0000001100000010", 2), tonumber(addAug[2], 2), tonumber(addAug[3], 2), tonumber(addAug[4] .. addAug[5], 2))
+        elseif addon == 2 then
+            player:updateEvent(tonumber(addAug[1] .. "0000001100000010", 2), tonumber(addAug[2] .. addAug[3], 2), tonumber(addAug[4], 2), tonumber(addAug[5], 2))
+        elseif addon == 3 then
+            player:updateEvent(tonumber(addAug[1] .. "0000001100000010", 2), tonumber(addAug[2] .. addAug[3], 2), tonumber(addAug[4], 2), tonumber(addAug[5], 2))
+        end -- Param placement is as follows (Aug1, Aug2 + Aug3, Aug4)
+    end
+end
+
+function intToBinary(x)
```

Comment:
Same as before: `scripts/zones/Lower_Jeuno/npcs/Treasure_Coffer.lua:633:61: (W113) accessing undefined variable intToBinary`, this could be `local`, and defined nearer the top of the file (think C-style declarations in a single file)